Feature/action controller

henrycity opened the pull request:
- Add a sheet at the end of the application.
- Click "Show Direction" to show the line.

danielcwilson committed:
Show alert and vibrate when user is "near" destination beacon

Currently can only be used once before crashing, needs debugging. However for the one time it works its pretty good eh

Commit:
Add the ActionController but need to change the naming

danielcwilson committed:
Fix crash on second navigate attempt

Also remove an unnecessary log

danielcwilson committed:
Merge branch 'gradient-buttons-with-swiftlint' into arrive-alert

I love programming in like 3 different branches at once

Commit:
Add action controller

Commit:
Add RoomCell.xib

danielcwilson committed:
Possibly fix meeting room sheet thing

No idea why it works but it seems to work now

Commit:
Working action controller

danielcwilson committed:
Merge remote-tracking branch 'origin/dev' into arrive-alert

Commit:
Merge remote-tracking branch 'origin/dev' into feature/action-controller

Commit:
Fix beacon.accuracy error

danielcwilson commented on MobileDevProject/Room.swift:
```diff
@@ -0,0 +1,195 @@
+    open lazy var title: UILabel = {
+        let title = UILabel(frame: CGRect.zero)
+        title.font = UIFont(name: "HelveticaNeue-Bold", size: 18)
+        title.text = "The Fast And ... The Furious Soundtrack Collection"
```
Placeholder values should be removed or at least seemingly relevant

danielcwilson commented on MobileDevProject/ViewController.swift:
```diff
@@ -111,7 +102,7 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
             for beacon in beaconInfo {
                 var buttonColour: UIColor
-                let colourAmount = (255 - (CGFloat(beacons[count].accuracy) * 40))
+                let colourAmount = (255 - (CGFloat(beacons[0].accuracy) * 40))
```
This should still be beacons[count] unless we're removing having all buttons change colour

danielcwilson commented on MobileDevProject/Room.swift:
```diff
@@ -0,0 +1,195 @@
+        return title
+    }()
+
+    open lazy var artist: UILabel = {
```
Artist is definitely not the variable name we need or want

danielcwilson commented on MobileDevProject/Room.swift:
```diff
@@ -0,0 +1,195 @@
+        return imageView
+    }()
+
+    open lazy var title: UILabel = {
```
Title works as a variable name but may be clearer if it was "room" or something similar

danielcwilson commented on MobileDevProject/Room.swift:
```diff
@@ -0,0 +1,195 @@
+
+    var title: String
+    var subtitle: String
+//    var image: UIImage
```
If we're definitely not using images remove the commented lines

henrycity replied:
At first, I am thinking about if we should have an image in the ActionController on the left. It would look nice if you can have a picture of the room. Don't know what about others think.

danielcwilson replied:
Ah okay, I'm not against that idea but implementation may be difficult

leenderteloff replied:
like a photo?

leenderteloff replied:
I like the idea, just to give the user an idea what the office looks like

danielcwilson reviewed (changes requested):
There might be things I'm missing in Room.swift but I don't quite understand it, so I assume its fine other than the comments I left

leenderteloff commented:
What about more information about the room? availability, space, amount of seats, etc

tvtri96 commented:
@leenderteloff We have to think about that together. Since in the app we show, it is not about the meeting room anymore.

vaguelycompetent commented:
In theory aye but we don't have access to that info. Currently rooms can be shown as busy or available but its just hardcoded

Commit:
Change name of variables and the index of beacons for the colorAmount

danielcwilson committed:
Fix index out of bounds error when calculating colour

Exact colours can be adjusted at another time with  group advice. 

Its very likely that I used the wrong condition so please test, I always mess things like that up 
Also I super disagree with lint i feel like if conditions should be surrounded with brackets but whatever

danielcwilson committed:
Less lines more better

Commit:
Change the color of arrow, circle and remove unnecceassry remove layer of circle

tvtri96 commented:
@leenderteloff : this branch please

Commit:
Remove circleShapeDrawn

Commit:
Add XLActionController as pod
